graylog2-web-interface/src/views/components/contexts/WidgetFocusProvider.test.tsx

```diff
@@ -80,4 +81,23 @@ describe('WidgetFocusProvider', () => {
     renderSUT();
     await screen.findByText('clack');
   });
+
+  it('should not set focused widget from url if the widget does not exist', async () => {
+    // @ts-ignore this is already mocked above.
+    useStore.mockReturnValue({
+      has: jest.fn(() => false),
+    });
+
+    useLocation.mockReturnValue({
+      pathname: '',
+      search: 'focused=clack',
+    });
+
+    renderSUT();
+    const clack = screen.findByText('clack');
+
+    waitFor(() => {
+      expect(clack).toBeEmptyDOMElement();
```

Comment:
Using `waitFor` here will not work. `clack` will always be the same, so it will either work the first time or never.

What are you testing here? If you want to check if focused widget id context does not have a widget id set, you could do:

```
await findByText('undefined');
```

Reply:
No, I needed to use `waitFor` here, since clack will never be undefined but a Promise with an empty Object. Only when I do a `waitFor` It will return as an `EmptyDomElement` :man_shrugging:

Reply:
The only reason why this works is that the test does not wait for `waitFor` to finish (because of a missing `await` or `return`). This way, `waitFor` generates a promise that will be executed in the background, but the test will just finish before it was resolved.